src/Mail/MailInterface.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace ZfrMail\Mail;
+
+/**
+ * Simple interface for a mail
+ *
+ * @author Michaël Gallego
+ */
+interface MailInterface
+{
+    /**
+     * Add a new "from" address and return a new mail
+     *
+     * @param  string $from
+     * @return MailInterface
+     */
+    public function withFrom(string $from): MailInterface;
```

Comment:
Okay I can do that, althoough PhpStorm is not really happy and says there is a mismatch between the return typehint (MailInterface) and what you indicate in the PHPDoc.